libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Classes.hs

```diff
@@ -531,11 +515,13 @@ data TxF era where
 unTxF :: TxF era -> Tx era
 unTxF (TxF _ x) = x
 
-instance PrettyA (TxF era) where
-  prettyA (TxF p tx) = pcTx p tx
+-- TODO: this feels iffy
+deriving instance Generic (TxF era)
```

Comment:
You can switch `TxF` to not use `GADT` syntax and then use the usual `Generic` deriving.
There is no reason why it should be using GADTs